oryx-app-mllib/src/test/java/com/cloudera/oryx/app/mllib/kmeans/KMeansEvaluationIT.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2015, Cloudera and Intel, Inc. All Rights Reserved.
+ *
+ * Cloudera, Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"). You may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for
+ * the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package com.cloudera.oryx.app.mllib.kmeans;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Lists;
+import org.dmg.pmml.PMML;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Tuple2;
+
+import com.cloudera.oryx.app.kmeans.ClusterInfo;
+import com.cloudera.oryx.app.kmeans.DistanceFn;
+import com.cloudera.oryx.app.kmeans.KMeansPMMLUtils;
+import com.cloudera.oryx.app.kmeans.SquaredDistanceFn;
+
+public class KMeansEvaluationIT extends AbstractKMeansIT {
+
+  private final static Logger log = LoggerFactory.getLogger(KMeansEvaluationIT.class);
+
+  private static final List<double[]> points =
+      Arrays.asList(new double[][]{
+          {1.0, 0.0}, {2.0, -2.0}, {2.0, 0.0},
+          {-2.0, 0.0}, {-0.5, -1.0}, {-0.5, 1.0}
+      });
+
+  private static List<ClusterInfo> clusters;
+  private static DistanceFn<double[]> distanceFn = new SquaredDistanceFn();
+
+  @Test
+  public void testEvalClustering() {
+    for (KMeansEvalStrategy evaluationStrategy : KMeansEvalStrategy.values()) {
+      switch (evaluationStrategy) {
+        case DAVIES_BOULDIN:
+          testDaviesBouldinIndexForClustering();
+          break;
+        case DUNN:
+          testDunnIndexForClustering();
+          break;
+        case SILHOUETTE:
+          testSilhouetteCoefficientForClustering();
+          break;
+        case SSE:
+          // do nothing
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown evaluation strategy " + evaluationStrategy);
+      }
+    }
+  }
+
+  public static void testDaviesBouldinIndexForClustering() {
+    KMeansEvaluation kMeansEvaluation = getKMeansEvaluationCriteria();
+    Map<Integer, Tuple2<Double, Long>> clusterSumDistancesAndCounts =
+        fetchClusterSumDistanceAndCounts(points);
+
+    double daviesBouldinIndex = kMeansEvaluation.daviesBouldinIndex(clusterSumDistancesAndCounts);
+    assertEquals(NUM_CLUSTERS, clusters.size());
+    log.info("Davies Bouldin Index for {} clusters: {}", clusters.size(), daviesBouldinIndex);
+    assertEquals(0.638, daviesBouldinIndex, 0.005);
```

Comment:
I would imagine these are pretty deterministic and so we should get a reliably exact figure?